core/src/main/java/nu/yona/server/exceptions/InvalidDataException.java

```diff
@@ -146,8 +148,23 @@ public static InvalidDataException invalidDate(DateTimeParseException exception,
 		return new InvalidDataException(exception, "error.invalid.date", date);
 	}
 
+	public static InvalidDataException dateTooEarly(LocalDateTime dateTime, LocalDateTime earliestPossibleDateTime)
+	{
+		return dateTooEarly(dateTime.toLocalDate(), earliestPossibleDateTime.toLocalDate());
+	}
+
+	public static InvalidDataException dateTooEarly(LocalDate date, LocalDate earliestPossibleDate)
+	{
+		return new InvalidDataException("error.invalid.date.too.early", date, earliestPossibleDate);
+	}
+
 	public static InvalidDataException missingRequestParameter(String name, String hint)
 	{
 		return new InvalidDataException("error.request.missing.request.parameter", name, hint);
 	}
+
+	public static InvalidDataException onlyAllowedOnTestServers(String hint)

```

Comment:
This looks more like an invalid request exception than an `InvalidDataException`